src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java

```diff
@@ -33,7 +33,7 @@ public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Task> lastShownList = model.getFilteredTaskList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        if (lastShownList == null || targetIndex.getZeroBased() >= lastShownList.size()) {
```

Comment:
No need to check null?

Reply:
needs because it fails some test case, I suspect storage is not working properly or something